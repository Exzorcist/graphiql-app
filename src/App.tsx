import GraphQLEditor from './components/GraphQLEditor/GraphQLEditor';
import ThemeProvider from './providers/ThemeProvider';

function App() {
  return (
    <ThemeProvider>
      <GraphQLEditor />
    </ThemeProvider>
  );
}

export default App;
