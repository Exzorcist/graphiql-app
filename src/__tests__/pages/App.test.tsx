import { screen, render } from '@testing-library/react';
import App from '@/App';
import { user } from '../test-utils';

test('Render App', async () => {
  render(<App />);
  expect(screen.getByText('GQL sandbox')).toBeInTheDocument();
  await user.click(screen.getByText('eng'));
  expect(screen.getByText('English')).toBeInTheDocument();
});
