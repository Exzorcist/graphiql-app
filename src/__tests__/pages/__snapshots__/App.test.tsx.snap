// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Render App 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div />
      <div
        class="min-h-screen flex flex-col"
      >
        <header
          class="bg-main py-3 md:py-4 text-white sticky top-0 transition-colors duration-500 z-10"
        >
          <div
            class="max-w-7xl mx-auto px-5 flex items-center justify-between gap-5"
          >
            <a
              class="flex items-center gap-4 text-lg sm:text-xl font-medium tracking-wider transition-colors
                    duration-300 hover:text-orange-200"
              href="/"
            >
              <div
                class="w-6 h-6"
              >
                <div>
                  <div
                    data-src="/src/assets/logo-graphql.svg"
                    src=""
                  />
                </div>
              </div>
              GQL sandbox
            </a>
            <div
              class="relative flex items-center gap-3"
            >
              <span
                aria-hidden="true"
                class="sm:hidden transition-colors duration-300 hover:text-orange-200 cursor-pointer"
              >
                <svg
                  aria-hidden="true"
                  class="w-8 h-8"
                  data-slot="icon"
                  fill="none"
                  stroke="currentColor"
                  stroke-width="1.5"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M3.75 6.75h16.5M3.75 12h16.5M12 17.25h8.25"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                </svg>
              </span>
              <div
                class="relative"
              >
                <div
                  aria-hidden="true"
                  class="flex gap-2 items-center cursor-pointer transition-colors duration-300 hover:text-orange-200"
                >
                  <span
                    class="hidden sm:block"
                  >
                    eng
                  </span>
                  <svg
                    aria-hidden="true"
                    class="w-8 h-8 sm:w-6 sm:h-6"
                    data-slot="icon"
                    fill="none"
                    stroke="currentColor"
                    stroke-width="1.5"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M12 21a9.004 9.004 0 0 0 8.716-6.747M12 21a9.004 9.004 0 0 1-8.716-6.747M12 21c2.485 0 4.5-4.03 4.5-9S14.485 3 12 3m0 18c-2.485 0-4.5-4.03-4.5-9S9.515 3 12 3m0 0a8.997 8.997 0 0 1 7.843 4.582M12 3a8.997 8.997 0 0 0-7.843 4.582m15.686 0A11.953 11.953 0 0 1 12 10.5c-2.998 0-5.74-1.1-7.843-2.918m15.686 0A8.959 8.959 0 0 1 21 12c0 .778-.099 1.533-.284 2.253m0 0A17.919 17.919 0 0 1 12 16.5c-3.162 0-6.133-.815-8.716-2.247m0 0A9.015 9.015 0 0 1 3 12c0-1.605.42-3.113 1.157-4.418"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                    />
                  </svg>
                </div>
                <div
                  class="absolute top-12 right-0 z-10 grid bg-white text-main text-base duration-300 sm:top-10 w-32 shadow rounded-md text-right overflow-hidden transition-all opacity-0 invisible"
                >
                  <span
                    aria-hidden="true"
                    class="py-2 px-3.5 cursor-pointer transition-colors duration-300 hover:bg-main/10 border-b border-neutral-200 last:border-b-0 bg-main/80 text-white pointer-events-none"
                  >
                    English
                  </span>
                  <span
                    aria-hidden="true"
                    class="py-2 px-3.5 cursor-pointer transition-colors duration-300 hover:bg-main/10 border-b border-neutral-200 last:border-b-0"
                  >
                    Русский
                  </span>
                  <span
                    aria-hidden="true"
                    class="py-2 px-3.5 cursor-pointer transition-colors duration-300 hover:bg-main/10 border-b border-neutral-200 last:border-b-0"
                  >
                    Беларускі
                  </span>
                </div>
              </div>
              <div
                class="absolute top-12 right-0 z-10 grid bg-white text-main text-base duration-300 py-1.5 min-w-[155px] w-32 shadow rounded-md text-right overflow-hidden transition-all opacity-0 invisible sm:static sm:!opacity-100 sm:!visible sm:bg-transparent sm:flex sm:text-white sm:shadow-none sm:gap-4 sm:p-0 sm:min-w-0 sm:w-full sm:!translate-y-0"
              >
                <span
                  class="hidden sm:block"
                >
                  |
                </span>
                <a
                  aria-current="page"
                  class="py-2 px-3.5 cursor-pointer transition-colors duration-300 hover:bg-main/10
                         border-b border-neutral-200 sm:border-b-0 sm:p-0 sm:hover:bg-transparent
                         sm:hover:text-orange-200 active"
                  href="/welcome"
                >
                  About us
                </a>
                <span
                  class="hidden sm:block"
                >
                  |
                </span>
                <a
                  class="py-2 px-3.5 cursor-pointer transition-colors duration-300 hover:bg-main/10
                         sm:p-0 sm:hover:bg-transparent sm:hover:text-orange-200"
                  href="/login"
                >
                  Login
                </a>
              </div>
            </div>
          </div>
        </header>
        <div
          class="flex-grow grid"
        >
          <section
            class="max-w-7xl mx-auto px-5 py-10 grid gap-8"
          >
            <h1
              class="text-4xl font-medium mt-8 mb-5 grid sm:flex justify-between gap-10 items-start"
            >
              Welcome to the GraphQL sandbox!
              <div
                class="flex gap-2 items-start flex-wrap text-center lg:flex-nowrap sm:whitespace-nowrap"
              >
                <a
                  class="bg-main hover:bg-main/80 font-medium rounded-lg text-sm w-full
                             px-8 py-2 text-white transition-colors duration-300 lg:w-auto"
                  href="/login"
                >
                  Sign In
                </a>
                <a
                  class="bg-main hover:bg-main/80 font-medium rounded-lg text-sm w-full
                             px-8 py-2 text-white transition-colors duration-300 lg:w-auto"
                  href="/registration"
                >
                  Sign Up
                </a>
              </div>
            </h1>
            <div
              class="text-center leading-8 text-neutral-700 italic border-2 border-main rounded-2xl
                   py-6 px-4 bg-main/10 mb-8"
            >
              No matter if you are an experienced developer or just starting out, our application will not leave you indifferent. Intuitive interface, localization and navigation, everything is done in the best traditions of React. Our application allows you to make GraphQL queries for any API. But before you start, don't forget to register. Also, for your convenience, we have added support for three languages: English, Belarusian, and Russian.
            </div>
            <h2
              class="text-center text-2xl font-medium max-w-xl mx-auto mb-5"
            >
              'Just Do It' is not just a team, it is a call to action that forms our own coding path.
            </h2>
            <div
              class="grid gap-6"
            >
              <div
                class="grid sm:grid-cols-[200px_1fr] gap-5 border-b border-neutral-300 pb-5 last:border-0 last:pb-0"
              >
                <img
                  alt="Avatar Ivan"
                  class="rounded-xl max-w-sm w-full mx-auto"
                  src="/src/assets/team/Ivan.jpg"
                />
                <div>
                  <h2
                    class="text-3xl font-medium mb-1"
                  >
                    Ivan
                  </h2>
                  <h3
                    class="text-neutral-500 italic mb-4 text-lg"
                  >
                    { Team leader | Frontend developer }
                  </h3>
                  <p
                    class="leading-7 tracking-wide mb-8"
                  >
                    He was involved in managing the team, creating navigation, basic elements and localization, and was on hand in difficult moments. An algorithm was also developed to prettifying query.
                  </p>
                  <a
                    aria-label="Link to gihub"
                    class="inline-flex gap-2 items-center transition-colors duration-300 hover:text-main"
                    href="https://github.com/Exzorcist"
                  >
                    <div
                      class="w-8 h-8"
                    >
                      <div>
                        <div
                          data-src="/src/assets/logo-github.svg"
                          src=""
                        />
                      </div>
                    </div>
                    Exzorcist
                  </a>
                </div>
              </div>
              <div
                class="grid sm:grid-cols-[200px_1fr] gap-5 border-b border-neutral-300 pb-5 last:border-0 last:pb-0"
              >
                <img
                  alt="Avatar Polina"
                  class="rounded-xl max-w-sm w-full mx-auto"
                  src="/src/assets/team/Polina.jpg"
                />
                <div>
                  <h2
                    class="text-3xl font-medium mb-1"
                  >
                    Polina
                  </h2>
                  <h3
                    class="text-neutral-500 italic mb-4 text-lg"
                  >
                    { Frontend developer | The soul of the team }
                  </h3>
                  <p
                    class="leading-7 tracking-wide mb-8"
                  >
                    She was engaged in the development of login and registration pages, firebase integration into the project, form validation. She also became the inspiration for the design of the application. Integrated logic for adding and autocompleting headers
                  </p>
                  <a
                    aria-label="Link to gihub"
                    class="inline-flex gap-2 items-center transition-colors duration-300 hover:text-main"
                    href="https://github.com/PolinaPolina94"
                  >
                    <div
                      class="w-8 h-8"
                    >
                      <div>
                        <div
                          data-src="/src/assets/logo-github.svg"
                          src=""
                        />
                      </div>
                    </div>
                    PolinaPolina94
                  </a>
                </div>
              </div>
              <div
                class="grid sm:grid-cols-[200px_1fr] gap-5 border-b border-neutral-300 pb-5 last:border-0 last:pb-0"
              >
                <img
                  alt="Avatar Marat"
                  class="rounded-xl max-w-sm w-full mx-auto"
                  src="/src/assets/team/Marat.jpg"
                />
                <div>
                  <h2
                    class="text-3xl font-medium mb-1"
                  >
                    Marat
                  </h2>
                  <h3
                    class="text-neutral-500 italic mb-4 text-lg"
                  >
                    { Frontend developer | The main sage of the team }
                  </h3>
                  <p
                    class="leading-7 tracking-wide mb-8"
                  >
                    Took responsibility for the most difficult part, namely GraphQL Editor. He helped to implement many useful and interesting solutions for simplifying life on the project.
                  </p>
                  <a
                    aria-label="Link to gihub"
                    class="inline-flex gap-2 items-center transition-colors duration-300 hover:text-main"
                    href="https://github.com/nowaylifer"
                  >
                    <div
                      class="w-8 h-8"
                    >
                      <div>
                        <div
                          data-src="/src/assets/logo-github.svg"
                          src=""
                        />
                      </div>
                    </div>
                    nowaylifer
                  </a>
                </div>
              </div>
            </div>
            <div
              class="text-center leading-8 text-neutral-700 italic border-2 border-main rounded-2xl
                   py-6 px-4 bg-main/10 mb-8 mt-10"
            >
              This application was created as part of the RS School React course. These are free online courses where everyone can learn and comprehend new technologies, regardless of age, professional employment and place of residence, everything is in your hands. You can find a link to the course below.
            </div>
          </section>
        </div>
        <footer>
          <div
            class="bg-main text-white"
          >
            <div
              class="flex gap-5 py-2 px-5 justify-between items-center max-w-7xl mx-auto"
            >
              <div
                class="flex gap-4 items-center"
              >
                <a
                  aria-label="GraphQL Logo github"
                  class="flex items-center gap-2 transition-colors duration-300 hover:text-orange-200"
                  href="https://github.com/Exzorcist/graphiql-app/tree/develop"
                  rel="noreferrer"
                  target="_blank"
                >
                  <div
                    class="w-10 h-10"
                  >
                    <div>
                      <div
                        data-src="/src/assets/logo-github.svg"
                        src=""
                      />
                    </div>
                  </div>
                  <span
                    class="hidden sm:block"
                  >
                    "Just Do It" team
                  </span>
                </a>
                <span
                  class="text-xl font-thin"
                >
                  |
                </span>
                <span
                  class="text-lg font-light flex gap-1.5"
                >
                  <span
                    class="hidden sm:block"
                  >
                    React
                  </span>
                  <span
                    class="font-medium"
                  >
                    2023
                  </span>
                  <span
                    class="hidden sm:block"
                  >
                    Q4
                  </span>
                </span>
              </div>
              <a
                aria-label="RSSchool Logo"
                class="transition-transform duration-300 hover:scale-110"
                href="https://rs.school/react/"
                rel="noreferrer"
                target="_blank"
              >
                <div
                  class="w-20 h-10"
                >
                  <div>
                    <div
                      data-src="/src/assets/logo-rsschool.svg"
                      src=""
                    />
                  </div>
                </div>
              </a>
            </div>
          </div>
        </footer>
      </div>
      <div
        class="fixed bottom-3 right-3 z-30 py-2 pr-5 pl-14 rounded-xl text-lg max-w-[280px] md:max-w-sm lg:max-w-lg transition-transform duration-500 delay-500 translate-x-[120%] border bg-blue-100 text-blue-700 border-blue-700"
        data-element="global-message"
      >
        <svg
          aria-hidden="true"
          class="w-6 h-6 absolute top-2.5 left-5"
          data-slot="icon"
          fill="none"
          stroke="currentColor"
          stroke-width="1.5"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M10.34 15.84c-.688-.06-1.386-.09-2.09-.09H7.5a4.5 4.5 0 1 1 0-9h.75c.704 0 1.402-.03 2.09-.09m0 9.18c.253.962.584 1.892.985 2.783.247.55.06 1.21-.463 1.511l-.657.38c-.551.318-1.26.117-1.527-.461a20.845 20.845 0 0 1-1.44-4.282m3.102.069a18.03 18.03 0 0 1-.59-4.59c0-1.586.205-3.124.59-4.59m0 9.18a23.848 23.848 0 0 1 8.835 2.535M10.34 6.66a23.847 23.847 0 0 0 8.835-2.535m0 0A23.74 23.74 0 0 0 18.795 3m.38 1.125a23.91 23.91 0 0 1 1.014 5.395m-1.014 8.855c-.118.38-.245.754-.38 1.125m.38-1.125a23.91 23.91 0 0 0 1.014-5.395m0-3.46c.495.413.811 1.035.811 1.73 0 .695-.316 1.317-.811 1.73m0-3.46a24.347 24.347 0 0 1 0 3.46"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
      </div>
    </div>
  </body>,
  "container": <div>
    <div />
    <div
      class="min-h-screen flex flex-col"
    >
      <header
        class="bg-main py-3 md:py-4 text-white sticky top-0 transition-colors duration-500 z-10"
      >
        <div
          class="max-w-7xl mx-auto px-5 flex items-center justify-between gap-5"
        >
          <a
            class="flex items-center gap-4 text-lg sm:text-xl font-medium tracking-wider transition-colors
                    duration-300 hover:text-orange-200"
            href="/"
          >
            <div
              class="w-6 h-6"
            >
              <div>
                <div
                  data-src="/src/assets/logo-graphql.svg"
                  src=""
                />
              </div>
            </div>
            GQL sandbox
          </a>
          <div
            class="relative flex items-center gap-3"
          >
            <span
              aria-hidden="true"
              class="sm:hidden transition-colors duration-300 hover:text-orange-200 cursor-pointer"
            >
              <svg
                aria-hidden="true"
                class="w-8 h-8"
                data-slot="icon"
                fill="none"
                stroke="currentColor"
                stroke-width="1.5"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M3.75 6.75h16.5M3.75 12h16.5M12 17.25h8.25"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
            </span>
            <div
              class="relative"
            >
              <div
                aria-hidden="true"
                class="flex gap-2 items-center cursor-pointer transition-colors duration-300 hover:text-orange-200"
              >
                <span
                  class="hidden sm:block"
                >
                  eng
                </span>
                <svg
                  aria-hidden="true"
                  class="w-8 h-8 sm:w-6 sm:h-6"
                  data-slot="icon"
                  fill="none"
                  stroke="currentColor"
                  stroke-width="1.5"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12 21a9.004 9.004 0 0 0 8.716-6.747M12 21a9.004 9.004 0 0 1-8.716-6.747M12 21c2.485 0 4.5-4.03 4.5-9S14.485 3 12 3m0 18c-2.485 0-4.5-4.03-4.5-9S9.515 3 12 3m0 0a8.997 8.997 0 0 1 7.843 4.582M12 3a8.997 8.997 0 0 0-7.843 4.582m15.686 0A11.953 11.953 0 0 1 12 10.5c-2.998 0-5.74-1.1-7.843-2.918m15.686 0A8.959 8.959 0 0 1 21 12c0 .778-.099 1.533-.284 2.253m0 0A17.919 17.919 0 0 1 12 16.5c-3.162 0-6.133-.815-8.716-2.247m0 0A9.015 9.015 0 0 1 3 12c0-1.605.42-3.113 1.157-4.418"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                </svg>
              </div>
              <div
                class="absolute top-12 right-0 z-10 grid bg-white text-main text-base duration-300 sm:top-10 w-32 shadow rounded-md text-right overflow-hidden transition-all opacity-0 invisible"
              >
                <span
                  aria-hidden="true"
                  class="py-2 px-3.5 cursor-pointer transition-colors duration-300 hover:bg-main/10 border-b border-neutral-200 last:border-b-0 bg-main/80 text-white pointer-events-none"
                >
                  English
                </span>
                <span
                  aria-hidden="true"
                  class="py-2 px-3.5 cursor-pointer transition-colors duration-300 hover:bg-main/10 border-b border-neutral-200 last:border-b-0"
                >
                  Русский
                </span>
                <span
                  aria-hidden="true"
                  class="py-2 px-3.5 cursor-pointer transition-colors duration-300 hover:bg-main/10 border-b border-neutral-200 last:border-b-0"
                >
                  Беларускі
                </span>
              </div>
            </div>
            <div
              class="absolute top-12 right-0 z-10 grid bg-white text-main text-base duration-300 py-1.5 min-w-[155px] w-32 shadow rounded-md text-right overflow-hidden transition-all opacity-0 invisible sm:static sm:!opacity-100 sm:!visible sm:bg-transparent sm:flex sm:text-white sm:shadow-none sm:gap-4 sm:p-0 sm:min-w-0 sm:w-full sm:!translate-y-0"
            >
              <span
                class="hidden sm:block"
              >
                |
              </span>
              <a
                aria-current="page"
                class="py-2 px-3.5 cursor-pointer transition-colors duration-300 hover:bg-main/10
                         border-b border-neutral-200 sm:border-b-0 sm:p-0 sm:hover:bg-transparent
                         sm:hover:text-orange-200 active"
                href="/welcome"
              >
                About us
              </a>
              <span
                class="hidden sm:block"
              >
                |
              </span>
              <a
                class="py-2 px-3.5 cursor-pointer transition-colors duration-300 hover:bg-main/10
                         sm:p-0 sm:hover:bg-transparent sm:hover:text-orange-200"
                href="/login"
              >
                Login
              </a>
            </div>
          </div>
        </div>
      </header>
      <div
        class="flex-grow grid"
      >
        <section
          class="max-w-7xl mx-auto px-5 py-10 grid gap-8"
        >
          <h1
            class="text-4xl font-medium mt-8 mb-5 grid sm:flex justify-between gap-10 items-start"
          >
            Welcome to the GraphQL sandbox!
            <div
              class="flex gap-2 items-start flex-wrap text-center lg:flex-nowrap sm:whitespace-nowrap"
            >
              <a
                class="bg-main hover:bg-main/80 font-medium rounded-lg text-sm w-full
                             px-8 py-2 text-white transition-colors duration-300 lg:w-auto"
                href="/login"
              >
                Sign In
              </a>
              <a
                class="bg-main hover:bg-main/80 font-medium rounded-lg text-sm w-full
                             px-8 py-2 text-white transition-colors duration-300 lg:w-auto"
                href="/registration"
              >
                Sign Up
              </a>
            </div>
          </h1>
          <div
            class="text-center leading-8 text-neutral-700 italic border-2 border-main rounded-2xl
                   py-6 px-4 bg-main/10 mb-8"
          >
            No matter if you are an experienced developer or just starting out, our application will not leave you indifferent. Intuitive interface, localization and navigation, everything is done in the best traditions of React. Our application allows you to make GraphQL queries for any API. But before you start, don't forget to register. Also, for your convenience, we have added support for three languages: English, Belarusian, and Russian.
          </div>
          <h2
            class="text-center text-2xl font-medium max-w-xl mx-auto mb-5"
          >
            'Just Do It' is not just a team, it is a call to action that forms our own coding path.
          </h2>
          <div
            class="grid gap-6"
          >
            <div
              class="grid sm:grid-cols-[200px_1fr] gap-5 border-b border-neutral-300 pb-5 last:border-0 last:pb-0"
            >
              <img
                alt="Avatar Ivan"
                class="rounded-xl max-w-sm w-full mx-auto"
                src="/src/assets/team/Ivan.jpg"
              />
              <div>
                <h2
                  class="text-3xl font-medium mb-1"
                >
                  Ivan
                </h2>
                <h3
                  class="text-neutral-500 italic mb-4 text-lg"
                >
                  { Team leader | Frontend developer }
                </h3>
                <p
                  class="leading-7 tracking-wide mb-8"
                >
                  He was involved in managing the team, creating navigation, basic elements and localization, and was on hand in difficult moments. An algorithm was also developed to prettifying query.
                </p>
                <a
                  aria-label="Link to gihub"
                  class="inline-flex gap-2 items-center transition-colors duration-300 hover:text-main"
                  href="https://github.com/Exzorcist"
                >
                  <div
                    class="w-8 h-8"
                  >
                    <div>
                      <div
                        data-src="/src/assets/logo-github.svg"
                        src=""
                      />
                    </div>
                  </div>
                  Exzorcist
                </a>
              </div>
            </div>
            <div
              class="grid sm:grid-cols-[200px_1fr] gap-5 border-b border-neutral-300 pb-5 last:border-0 last:pb-0"
            >
              <img
                alt="Avatar Polina"
                class="rounded-xl max-w-sm w-full mx-auto"
                src="/src/assets/team/Polina.jpg"
              />
              <div>
                <h2
                  class="text-3xl font-medium mb-1"
                >
                  Polina
                </h2>
                <h3
                  class="text-neutral-500 italic mb-4 text-lg"
                >
                  { Frontend developer | The soul of the team }
                </h3>
                <p
                  class="leading-7 tracking-wide mb-8"
                >
                  She was engaged in the development of login and registration pages, firebase integration into the project, form validation. She also became the inspiration for the design of the application. Integrated logic for adding and autocompleting headers
                </p>
                <a
                  aria-label="Link to gihub"
                  class="inline-flex gap-2 items-center transition-colors duration-300 hover:text-main"
                  href="https://github.com/PolinaPolina94"
                >
                  <div
                    class="w-8 h-8"
                  >
                    <div>
                      <div
                        data-src="/src/assets/logo-github.svg"
                        src=""
                      />
                    </div>
                  </div>
                  PolinaPolina94
                </a>
              </div>
            </div>
            <div
              class="grid sm:grid-cols-[200px_1fr] gap-5 border-b border-neutral-300 pb-5 last:border-0 last:pb-0"
            >
              <img
                alt="Avatar Marat"
                class="rounded-xl max-w-sm w-full mx-auto"
                src="/src/assets/team/Marat.jpg"
              />
              <div>
                <h2
                  class="text-3xl font-medium mb-1"
                >
                  Marat
                </h2>
                <h3
                  class="text-neutral-500 italic mb-4 text-lg"
                >
                  { Frontend developer | The main sage of the team }
                </h3>
                <p
                  class="leading-7 tracking-wide mb-8"
                >
                  Took responsibility for the most difficult part, namely GraphQL Editor. He helped to implement many useful and interesting solutions for simplifying life on the project.
                </p>
                <a
                  aria-label="Link to gihub"
                  class="inline-flex gap-2 items-center transition-colors duration-300 hover:text-main"
                  href="https://github.com/nowaylifer"
                >
                  <div
                    class="w-8 h-8"
                  >
                    <div>
                      <div
                        data-src="/src/assets/logo-github.svg"
                        src=""
                      />
                    </div>
                  </div>
                  nowaylifer
                </a>
              </div>
            </div>
          </div>
          <div
            class="text-center leading-8 text-neutral-700 italic border-2 border-main rounded-2xl
                   py-6 px-4 bg-main/10 mb-8 mt-10"
          >
            This application was created as part of the RS School React course. These are free online courses where everyone can learn and comprehend new technologies, regardless of age, professional employment and place of residence, everything is in your hands. You can find a link to the course below.
          </div>
        </section>
      </div>
      <footer>
        <div
          class="bg-main text-white"
        >
          <div
            class="flex gap-5 py-2 px-5 justify-between items-center max-w-7xl mx-auto"
          >
            <div
              class="flex gap-4 items-center"
            >
              <a
                aria-label="GraphQL Logo github"
                class="flex items-center gap-2 transition-colors duration-300 hover:text-orange-200"
                href="https://github.com/Exzorcist/graphiql-app/tree/develop"
                rel="noreferrer"
                target="_blank"
              >
                <div
                  class="w-10 h-10"
                >
                  <div>
                    <div
                      data-src="/src/assets/logo-github.svg"
                      src=""
                    />
                  </div>
                </div>
                <span
                  class="hidden sm:block"
                >
                  "Just Do It" team
                </span>
              </a>
              <span
                class="text-xl font-thin"
              >
                |
              </span>
              <span
                class="text-lg font-light flex gap-1.5"
              >
                <span
                  class="hidden sm:block"
                >
                  React
                </span>
                <span
                  class="font-medium"
                >
                  2023
                </span>
                <span
                  class="hidden sm:block"
                >
                  Q4
                </span>
              </span>
            </div>
            <a
              aria-label="RSSchool Logo"
              class="transition-transform duration-300 hover:scale-110"
              href="https://rs.school/react/"
              rel="noreferrer"
              target="_blank"
            >
              <div
                class="w-20 h-10"
              >
                <div>
                  <div
                    data-src="/src/assets/logo-rsschool.svg"
                    src=""
                  />
                </div>
              </div>
            </a>
          </div>
        </div>
      </footer>
    </div>
    <div
      class="fixed bottom-3 right-3 z-30 py-2 pr-5 pl-14 rounded-xl text-lg max-w-[280px] md:max-w-sm lg:max-w-lg transition-transform duration-500 delay-500 translate-x-[120%] border bg-blue-100 text-blue-700 border-blue-700"
      data-element="global-message"
    >
      <svg
        aria-hidden="true"
        class="w-6 h-6 absolute top-2.5 left-5"
        data-slot="icon"
        fill="none"
        stroke="currentColor"
        stroke-width="1.5"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M10.34 15.84c-.688-.06-1.386-.09-2.09-.09H7.5a4.5 4.5 0 1 1 0-9h.75c.704 0 1.402-.03 2.09-.09m0 9.18c.253.962.584 1.892.985 2.783.247.55.06 1.21-.463 1.511l-.657.38c-.551.318-1.26.117-1.527-.461a20.845 20.845 0 0 1-1.44-4.282m3.102.069a18.03 18.03 0 0 1-.59-4.59c0-1.586.205-3.124.59-4.59m0 9.18a23.848 23.848 0 0 1 8.835 2.535M10.34 6.66a23.847 23.847 0 0 0 8.835-2.535m0 0A23.74 23.74 0 0 0 18.795 3m.38 1.125a23.91 23.91 0 0 1 1.014 5.395m-1.014 8.855c-.118.38-.245.754-.38 1.125m.38-1.125a23.91 23.91 0 0 0 1.014-5.395m0-3.46c.495.413.811 1.035.811 1.73 0 .695-.316 1.317-.811 1.73m0-3.46a24.347 24.347 0 0 1 0 3.46"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </svg>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
